mongo/open.go

```diff
@@ -127,26 +127,27 @@ func DialInfo(info Info, opts DialOpts) (*mgo.DialInfo, error) {
 	tlsConfig.RootCAs = pool
 	tlsConfig.ServerName = "juju-mongodb"
 
-	dial := func(addr net.Addr) (net.Conn, error) {
-		c, err := net.Dial("tcp", addr.String())
+	dial := func(server *mgo.ServerAddr) (net.Conn, error) {
+		addr := server.TCPAddr().String()
+		c, err := net.DialTimeout("tcp", addr, opts.Timeout)
 		if err != nil {
-			logger.Debugf("connection failed, will retry: %v", err)
+			logger.Errorf("mongodb connection failed, will retry: %v", err)
```

Comment:
s/error/warning

errors are supposed to be something a user acts on IIRC. the fact that it says "will retry" implies warning is sufficient